compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1216,7 +1216,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         dest_offset: Size,
         size: Size,
     ) -> bool {
-        let overlaps = |a, b| a <= b && b < a + size;
+        let overlaps = |a, b| a <= b && b - a < size;
```

Comment:
Please add a comment explaining why the subtraction cannot overflow.

Reply:
we could rewrite this as `a.checked_sub(b).map_or(false, |s| s < size)`

Reply:
Ok, I think we can just comment that ```since `a <= b`, `b - a` will not overflows ```